collie/model/hybrid_pretrained_matrix_factorization.py

```diff
@@ -114,37 +125,100 @@ def __init__(self,
-                item_metadata = torch.from_numpy(item_metadata)
-
-            item_metadata = item_metadata.float()
+            if item_metadata is None and user_metadata is None:
+                raise ValueError(
+                    'Must provide item metadata and/or user_metadata for ``HybridPretrainedModel``.'
+                )
 
-            item_metadata_num_cols = item_metadata.shape[1]
+            if item_metadata is not None:
+                if isinstance(item_metadata, pd.DataFrame):
+                    item_metadata = torch.from_numpy(item_metadata.to_numpy())
+                elif isinstance(item_metadata, np.ndarray):
+                    item_metadata = torch.from_numpy(item_metadata)
+                item_metadata = item_metadata.float()
+                item_metadata_num_cols = item_metadata.shape[1]
+
+            if user_metadata is not None:
+                if isinstance(user_metadata, pd.DataFrame):
+                    user_metadata = torch.from_numpy(user_metadata.to_numpy())
+                elif isinstance(user_metadata, np.ndarray):
+                    user_metadata = torch.from_numpy(user_metadata)
+                user_metadata = user_metadata.float()
+                user_metadata_num_cols = user_metadata.shape[1]
 
         super().__init__(**get_init_arguments(),
-                         item_metadata_num_cols=item_metadata_num_cols)
+                         item_metadata_num_cols=item_metadata_num_cols,
+                         user_metadata_num_cols=user_metadata_num_cols)
 
     __doc__ = merge_docstrings(BasePipeline, __doc__, __init__)
 
     def _move_any_external_data_to_device(self):
-        """Move item metadata to the device before training."""
-        self.item_metadata = self.item_metadata.to(self.device)
+        """Move item and/or user metadata to the device before training."""
+        if self.item_metadata is not None:
+            self.item_metadata = self.item_metadata.to(self.device)
+        if self.user_metadata is not None:
+            self.user_metadata = self.user_metadata.to(self.device)
 
     def _load_model_init_helper(self, load_model_path: str, map_location: str, **kwargs) -> None:
-        self.item_metadata = (
-            joblib.load(os.path.join(load_model_path, 'metadata.pkl'))
-        )
+        try:
+            self.item_metadata = (
+                joblib.load(os.path.join(load_model_path, 'item_metadata.pkl'))
+            )
+        except FileNotFoundError:
+            warnings.warn('`item_metadata.pkl` not found')
+        try:
+            self.user_metadata = (
+                joblib.load(os.path.join(load_model_path, 'user_metadata.pkl'))
+            )
+        except FileNotFoundError:
+            warnings.warn('`user_metadata.pkl` not found')
         super()._load_model_init_helper(load_model_path=os.path.join(load_model_path, 'model.pth'),
                                         map_location=map_location)
 
+    def _configure_metadata_layers(
```

Comment:
@nathancooperjones I changed your version of _configure_metadata_layers to simply take the `metadata_type` instead of taking two string parameters and wanted your thoughts. I can replace whichever version you like better so that `hybrid` and `hybrid_pretrained` are identical